frontend/src/api/mockAPI/assignments.js

```diff
@@ -316,12 +317,21 @@ class ActiveOffer extends MockAPIController {
                 )}`
             );
         }
-        return this.find(
+
+        const newOffer = this.find(
             this.create({
                 assignment_id: matchingAssignment.id,
-                status: "pending"
+                status: "provisional"
             })
         );
+
+        if (!this.data.offer) {
+            this.data.offers = [newOffer];
+        } else {
+            this.data.offer.push(newOffer);
+        }
```

Comment:
I'm not sure how you feel about boolean short circuit operations / oneliners with regards to readability, but you could write this as:
`this.data.offer ? this.data.offer.push(newOffer) : this.data.offers = [newOffer]`

Reply:
I'm very against that. It makes it *much* harder to read. The ternary operator can be used for setting variables (e.g. `x = foo ? "yes" : "no"`), but shouldn't be used for code that has side effects.